woocommerce/woocommerce-gutenberg-products-block
# Product Gallery > Prevent error when "layout" is missing in block attributes

## What
An error is currently occurring when adding the Product Gallery to the page and visiting a variable product page.

Fixes #10783 

## Why
This PR fixes the bug and make the block to work correctly with variable products.



## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. From your WordPress dashboard, go to Appearance > Themes. Make sure you have a block-based theme installed and activated. If not, you can install one from the Add New option. Block-based themes include "Twenty-twenty Two," "Twenty-twenty Three," etc.
2. On the left-hand side menu, click on Appearance > Editor > Templates
3. Find and select the 'Single Product' template from the list.
4. When the Classic Product Template renders, click on Transform into Blocks. This will transform the Classic template in a block template if you haven't done it before.
5. Inside the Page editor, click on the '+' button, usually found at the top left of the editing space or within the content area itself, to add a new block.
6. In the block library that pops up, search for the 'Product Gallery' block. Click on it to add the block to the template.
7. Check that the Previous/Next buttons block is displayed on the Editor.
8. Click on Save;
9. Visit a variable product and make sure no error is displayed.


## Screenshots or screencast



| Before | After |
| ------ | ----- |
| ![CleanShot 2023-08-30 at 17 11 39@2x](https://github.com/woocommerce/woocommerce-blocks/assets/20469356/86f8e3a4-5688-4087-a5cd-59bce3218d87) | <img width="1214" alt="CleanShot 2023-08-30 at 17 34 34@2x" src="https://github.com/woocommerce/woocommerce-blocks/assets/20469356/ad16c1a2-c225-4a70-af38-ec263d1548d3"> |

## WooCommerce Visibility


Required:
